src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -475,9 +475,15 @@ private void AppendLineNoPromptInjection(ITextBuffer buffer)
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.TypeChar"/>.</summary>
-            internal void TypeChar(char typedChar)
+            public void TypeChar(char typedChar)
             {
-                InsertText(typedChar.ToString());
+                using (var transaction = UndoHistory?.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
Actually, I just remember another (bigger) reason why I implemented those operations in Interactive Window: the undo history is associated with input text buffer instead of the entire TextView of the window (both editable and readonly buffers), which doesn't work well with `EditorOperations` and how it handles undo/redo's.